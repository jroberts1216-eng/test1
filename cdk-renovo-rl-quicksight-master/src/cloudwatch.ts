import {
  aws_cloudwatch as cloudwatch,
  Stack,
  StackProps,
} from 'aws-cdk-lib';
import { Construct } from 'constructs';

/**
 * Properties for QuickSight CloudWatch Dashboard
 */
export interface QuickSightCloudWatchDashboardProps extends StackProps {
  /**
   * QuickSight Dashboard ID to monitor
   */
  quickSightDashboardId: string;

  /**
   * QuickSight Dataset ID for ingestion metrics
   */
  datasetId: string;

  /**
   * CloudWatch Dashboard Name
   */
  dashboardName: string;
}

/**
 * Stack that creates a CloudWatch Dashboard for QuickSight metrics
 */
export class QuickSightCloudWatchDashboard extends Stack {
  constructor(scope: Construct, id: string, props: QuickSightCloudWatchDashboardProps) {
    super(scope, id, props);

    // Create CloudWatch Dashboard
    const dashboard = new cloudwatch.Dashboard(this, 'QuickSightMetricsDashboard', {
      dashboardName: props.dashboardName,
    });

    const dashboardId = props.quickSightDashboardId;
    const datasetId = props.datasetId;

    // SPICE Capacity Metrics (Aggregate)
    const spiceCapacityLimitWidget = new cloudwatch.GraphWidget({
      title: 'SPICE Capacity Limit',
      left: [
        new cloudwatch.Metric({
          namespace: 'AWS/QuickSight',
          metricName: 'SPICECapacityLimitInMB',
          statistic: 'Average',
          unit: cloudwatch.Unit.MEGABYTES,
        }),
      ],
      width: 8,
    });

    const spiceCapacityUsedWidget = new cloudwatch.GraphWidget({
      title: 'SPICE Capacity Used',
      left: [
        new cloudwatch.Metric({
          namespace: 'AWS/QuickSight',
          metricName: 'SPICECapacityConsumedInMB',
          statistic: 'Average',
          unit: cloudwatch.Unit.MEGABYTES,
        }),
      ],
      width: 8,
    });

    const spiceCapacityUtilizationWidget = new cloudwatch.GraphWidget({
      title: 'SPICE Capacity Utilization (%)',
      left: [
        new cloudwatch.MathExpression({
          expression: '(consumed / limit) * 100',
          usingMetrics: {
            consumed: new cloudwatch.Metric({
              namespace: 'AWS/QuickSight',
              metricName: 'SPICECapacityConsumedInMB',
              statistic: 'Average',
            }),
            limit: new cloudwatch.Metric({
              namespace: 'AWS/QuickSight',
              metricName: 'SPICECapacityLimitInMB',
              statistic: 'Average',
            }),
          },
          label: 'SPICE Utilization (%)',
        }),
      ],
      width: 8,
    });

    // Data Ingestion Metrics
    const ingestionInvocationCountWidget = new cloudwatch.GraphWidget({
      title: `Ingestion Invocation Count - Dataset ${datasetId}`,
      left: [
        new cloudwatch.Metric({
          namespace: 'AWS/QuickSight',
          metricName: 'IngestionInvocationCount',
          dimensionsMap: { DatasetId: datasetId },
          statistic: 'Sum',
        }),
      ],
      width: 12,
    });

    const ingestionRowCountWidget = new cloudwatch.GraphWidget({
      title: `Ingestion Row Count - Dataset ${datasetId}`,
      left: [
        new cloudwatch.Metric({
          namespace: 'AWS/QuickSight',
          metricName: 'IngestionRowCount',
          dimensionsMap: { DatasetId: datasetId },
          statistic: 'Sum',
        }),
      ],
      width: 12,
    });

    const ingestionErrorCountWidget = new cloudwatch.GraphWidget({
      title: `Ingestion Error Count - Dataset ${datasetId}`,
      left: [
        new cloudwatch.Metric({
          namespace: 'AWS/QuickSight',
          metricName: 'IngestionErrorCount',
          dimensionsMap: { DatasetId: datasetId },
          statistic: 'Sum',
        }),
      ],
      width: 12,
    });

    const ingestionLatencyWidget = new cloudwatch.GraphWidget({
      title: `Ingestion Latency - Dataset ${datasetId}`,
      left: [
        new cloudwatch.Metric({
          namespace: 'AWS/QuickSight',
          metricName: 'IngestionLatency',
          dimensionsMap: { DatasetId: datasetId },
          statistic: 'Average',
          unit: cloudwatch.Unit.MILLISECONDS,
        }),
      ],
      width: 12,
    });

    // Performance and Availability Metrics
    const dashboardViewLoadTimeWidget = new cloudwatch.GraphWidget({
      title: `Dashboard View Load Time - Dashboard ${dashboardId}`,
      left: [
        new cloudwatch.Metric({
          namespace: 'AWS/QuickSight',
          metricName: 'DashboardViewLoadTime',
          dimensionsMap: { DashboardId: dashboardId },
          statistic: 'Average',
          unit: cloudwatch.Unit.MILLISECONDS,
        }),
      ],
      width: 12,
    });

    const visualLoadTimeWidget = new cloudwatch.GraphWidget({
      title: `Visual Load Time - Dashboard ${dashboardId}`,
      left: [
        new cloudwatch.Metric({
          namespace: 'AWS/QuickSight',
          metricName: 'VisualLoadTime',
          dimensionsMap: { DashboardId: dashboardId },
          statistic: 'Average',
          unit: cloudwatch.Unit.MILLISECONDS,
        }),
      ],
      width: 12,
    });

    const dashboardViewCountWidget = new cloudwatch.GraphWidget({
      title: `Dashboard View Count - Dashboard ${dashboardId}`,
      left: [
        new cloudwatch.Metric({
          namespace: 'AWS/QuickSight',
          metricName: 'DashboardViewCount',
          dimensionsMap: { DashboardId: dashboardId },
          statistic: 'Sum',
          unit: cloudwatch.Unit.COUNT,
        }),
      ],
      width: 12,
    });

    const visualLoadErrorCountWidget = new cloudwatch.GraphWidget({
      title: `Visual Load Error Count - Dashboard ${dashboardId}`,
      left: [
        new cloudwatch.Metric({
          namespace: 'AWS/QuickSight',
          metricName: 'VisualLoadErrorCount',
          dimensionsMap: { DashboardId: dashboardId },
          statistic: 'Sum',
          unit: cloudwatch.Unit.COUNT,
        }),
      ],
      width: 12,
    });

    // Create section headers with dashboard and dataset IDs
    const performanceHeader = new cloudwatch.TextWidget({
      markdown: `### QuickSight Dashboard Performance Metrics\n**Dashboard ID: ${dashboardId}**`,
      width: 24,
      height: 2,
    });

    const ingestionHeader = new cloudwatch.TextWidget({
      markdown: `### QuickSight Data Ingestion Metrics\n**Dataset ID: ${datasetId}**`,
      width: 24,
      height: 2,
    });

    const capacityHeader = new cloudwatch.TextWidget({
      markdown: '### QuickSight SPICE Capacity Metrics\n**Account Level Metrics**',
      width: 24,
      height: 2,
    });

    // Add all widgets to the dashboard in the new order with section headers
    dashboard.addWidgets(
      // Performance metrics section (dashboard related)
      performanceHeader,
      dashboardViewLoadTimeWidget,
      visualLoadTimeWidget,
      dashboardViewCountWidget,
      visualLoadErrorCountWidget,

      // Ingestion metrics section (dataset related)
      ingestionHeader,
      ingestionInvocationCountWidget,
      ingestionRowCountWidget,
      ingestionErrorCountWidget,
      ingestionLatencyWidget,

      // SPICE capacity metrics section (aggregate)
      capacityHeader,
      spiceCapacityLimitWidget,
      spiceCapacityUsedWidget,
      spiceCapacityUtilizationWidget,
    );
  }
}
