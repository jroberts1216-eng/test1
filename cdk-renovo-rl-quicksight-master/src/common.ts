import {
  DmsSelectionRuleProps,
  DmsTransformationRuleProps,
} from './dms';

export enum Accounts {
  SERVICES = '719035735300',
  RLDEV = '462159781845',
  RLPROD = '416804266253',
};

export enum VpcIds {
  SERVICES = 'vpc-0860cddb8bcdc6fb9',
};

export enum RenovoLiveEnv {
  DEV = 'dev',
  PROD = 'prod',
};

export enum RenovoLiveDBSchemas {
  DBO = 'dbo',
  ANY = '%',
};

export enum RLDBEndpoints {
  DEV = 'app-dev-rl-renovolive-com-2.cbbvzhqawqdz.us-east-1.rds.amazonaws.com',
  PROD = 'app-prod-rl-renovolive-com.c0ztckwal0sp.us-east-1.rds.amazonaws.com',
}

export enum RLDBCredSecretArns {
  DEV = 'arn:aws:secretsmanager:us-east-1:719035735300:secret:RenovoLive/Dev/QuickSight/DataSource/RLDBCreds-A8VVA3',
  PROD = 'arn:aws:secretsmanager:us-east-1:719035735300:secret:RenovoLive/Prod/QuickSight/DataSource/RLDBCreds-sAq870',
}

export const DataSetReadOnlyPermissions = [
  'quicksight:DescribeDataSet',
  'quicksight:DescribeDataSetPermissions',
  'quicksight:PassDataSet',
  'quicksight:DescribeIngestion',
  'quicksight:ListIngestions'
]

export const DataSetReadWritePermissions = [
  'quicksight:DescribeDataSet',
  'quicksight:DescribeDataSetPermissions',
  'quicksight:PassDataSet',
  'quicksight:DescribeIngestion',
  'quicksight:ListIngestions',
  'quicksight:UpdateDataSet',
  'quicksight:DeleteDataSet',
  'quicksight:CreateIngestion',
  'quicksight:CancelIngestion',
  'quicksight:UpdateDataSetPermissions'
]


/**
 * The details of a table for ingestion
 * 
 * This includes the DMS related selection and transformation
 * rules that will be applied to the table you give for
 * tableName
 */
export interface TableDetails {
  /**
   * The name of the table
   */
  readonly tableName: string;
  /**
   * The table schema
   */
  readonly schema: RenovoLiveDBSchemas;
  /**
   * The primary key of the table
   */
  readonly primaryKey: string;
  /**
   * The selection rules for the table
   */
  readonly selectionRules?: DmsSelectionRuleProps[];
  /**
   * The transformation rules for the table
   */
  readonly transformationRules?: DmsTransformationRuleProps[];
}

/**
 * The details of a database for ingestion
 * 
 * This includes all table details related to this database
 * and the DMS related selection and transformation rules for
 * those tables as well as any additional rules that apply
 * to the database as a whole such as wildcarded transformation
 * rules
 */
export interface DatabaseDetails {
  /**
   * The name of the database
   */
  readonly databaseName: string;
  /**
   * The tables that will be loaded from the database
   */
  readonly tables: TableDetails[];
  /**
   * Additional selection rules for the database that
   * don't apply to a specific table
   */
  readonly selectionRules?: DmsSelectionRuleProps[];
  /**
   * Additional transformation rules for the database that
   * don't apply to a specific table
   */
  readonly transformationRules?: DmsTransformationRuleProps[];
}
