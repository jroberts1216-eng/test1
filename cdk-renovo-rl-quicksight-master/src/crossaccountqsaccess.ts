import {
  aws_iam as iam,
  Stack,
  StackProps,
} from 'aws-cdk-lib';
import { NagSuppressions } from 'cdk-nag';
import { Construct } from 'constructs';

export interface RenovoLiveQuickSightCrossAccountAccessStackProps extends StackProps {
  /**
   * The account IDs that will be granted access to actions
   * in the QuickSight account
   */
  readonly accountIds: string[];
  /**
   * The external ID that will be used to assume the role
   */
  readonly externalId: string;
  /**
   * The name of the role that will be created
   */
  readonly roleName: string;
}

export class RenovoLiveQuickSightCrossAccountAccessStack extends Stack {
  constructor(scope: Construct, id: string, props: RenovoLiveQuickSightCrossAccountAccessStackProps) {
    super(scope, id, props);

    const role = new iam.Role(this, 'QuickSightCrossAccountRole', {
      roleName: props.roleName,
      assumedBy: new iam.PrincipalWithConditions(
        new iam.CompositePrincipal(
          ...props.accountIds.map(accountId => new iam.AccountPrincipal(accountId)),
        ),
        {
          StringEquals: {
            'sts:ExternalId': props.externalId,
          },
        },
      ),
    });

    /**
     * TODO: Pending guidance on how we can limit RenovoLive to generating embed URLs
     * for specific dashboards only based on their usage being intended for RenovoLive.
     * As this currently stands applications or users with access to assume a role with
     * this policy can generate embed URLs for any user in the QuickSight account for any
     * dashboard they have access to.
     */
    const quickSightPolicy = new iam.ManagedPolicy(this, 'QuickSightCrossAccountPolicy', {
      description: 'Policy to be used with cross-account access to QuickSight',
      statements: [
        new iam.PolicyStatement({
          sid: 'Default',
          effect: iam.Effect.ALLOW,
          actions: [
            'quicksight:ListUsers',
            'quicksight:DescribeUser',
            'quicksight:RegisterUser',
            'quicksight:ListDashboards',
            'quicksight:DescribeDashboard',
            'quicksight:ListDashboardVersions',
            'quicksight:GetSessionEmbedUrl',
            'quicksight:CreateGroupMembership',
            'quicksight:DeleteGroupMembership',
            'quicksight:DeleteUser',
          ],
          resources: ['*'],
        }),
        new iam.PolicyStatement({
          sid: 'AllowEmbedForRegisteredUser',
          effect: iam.Effect.ALLOW,
          actions: [
            'quicksight:GenerateEmbedUrlForRegisteredUser',
            'quicksight:GetAuthCode',
            'quicksight:SearchDashboards',
          ],
          resources: [
            `arn:aws:quicksight:${Stack.of(this).region}:${Stack.of(this).account}:user/*`,
          ],
        }),
        new iam.PolicyStatement({
          sid: 'AllowSearchDashboards',
          effect: iam.Effect.ALLOW,
          actions: [
            'quicksight:SearchDashboards',
          ],
          resources: [
            `arn:aws:quicksight:${Stack.of(this).region}:${Stack.of(this).account}:dashboard/*`,
          ],
        }),
      ],
    });

    NagSuppressions.addResourceSuppressions(quickSightPolicy, [
      {
        id: 'AwsSolutions-IAM5',
        reason: 'This policy is intended to be used with cross-account access to QuickSight and the wildcarded permissions are necessary for the use case. \
        See role policy guidance here: https://aws.amazon.com/blogs/modernizing-with-aws/embedding-amazon-quicksight-analytics-in-net-applications/',
      },
    ]);

    role.addManagedPolicy(quickSightPolicy);
  }
}

export interface QuickSightCrossAccountRoleAssumptionPolicyProps extends StackProps {
  /**
   * The role arn that will be assumable
   */
  readonly roleArn: string;
  /**
   * The name of the policy
   */
  readonly policyName: string;
};

export class QuickSightCrossAccountRoleAssumptionPolicy extends Stack {
  constructor(scope: Construct, id: string, props: QuickSightCrossAccountRoleAssumptionPolicyProps) {
    super(scope, id, props);

    new iam.ManagedPolicy(this, 'QuickSightCrossAccountRoleAssumptionPolicy', {
      managedPolicyName: props.policyName,
      description: 'Policy to be used to assume the QuickSight cross-account role',
      statements: [
        new iam.PolicyStatement({
          sid: 'AssumeRole',
          effect: iam.Effect.ALLOW,
          actions: [
            'sts:AssumeRole',
          ],
          resources: [
            props.roleArn,
          ],
        }),
      ],
    });
  }
}
