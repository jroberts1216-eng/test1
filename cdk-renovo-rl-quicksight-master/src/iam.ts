import {
  aws_iam as iam,
  CfnOutput,
  Stack,
  StackProps,
} from 'aws-cdk-lib';
import { Construct } from 'constructs';

export interface CloudWatchDashboardAccessStackProps extends StackProps {
  /**
   * Optional policy name. If not provided, a name will be generated.
   */
  policyName?: string;
}

/**
 * A stack that creates a managed policy granting access to CloudWatch dashboards and their widgets
 */
export class CloudWatchDashboardAccessStack extends Stack {
  /**
   * The managed policy created by this stack
   */
  public readonly dashboardAccessPolicy: iam.ManagedPolicy;

  constructor(scope: Construct, id: string, props: CloudWatchDashboardAccessStackProps) {
    super(scope, id, props);

    // Create the managed policy
    this.dashboardAccessPolicy = new iam.ManagedPolicy(this, 'CloudWatchDashboardAccess', {
      managedPolicyName: props.policyName,
      description: 'Grants access to CloudWatch dashboards and the widgets they contain',
      statements: [
        new iam.PolicyStatement({
          effect: iam.Effect.ALLOW,
          actions: [
            'cloudwatch:GetDashboard',
            'cloudwatch:ListDashboards',
            'cloudwatch:GetMetricData',
            'cloudwatch:GetMetricStatistics',
            'cloudwatch:DescribeAlarms',
            'cloudwatch:GetMetricWidgetImage',
          ],
          resources: ['*'],
        }),
      ],
    });

    // Output the ARN of the created policy
    new CfnOutput(this, 'DashboardAccessPolicyArn', {
      value: this.dashboardAccessPolicy.managedPolicyArn,
      description: 'ARN of the CloudWatch Dashboard Access Policy',
      exportName: `${this.stackName}-DashboardAccessPolicyArn`,
    });
  }
}
