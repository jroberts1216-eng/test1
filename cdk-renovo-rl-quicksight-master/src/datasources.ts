import {
  aws_quicksight as quicksight,
  Stack,
  StackProps,
  aws_redshiftserverless as redshift,
} from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { RenovoLiveEnv, RLDBCredSecretArns, RLDBEndpoints } from './common';

export interface RenovoLiveDataSourcesProps extends StackProps {
  /**
   * The RenovoLive environment
   */
  readonly renovoLiveEnv: RenovoLiveEnv;
  /**
   * The Redshift workgroup to connect to
   */
  readonly redshiftWorkgroup: redshift.CfnWorkgroup;
  /**
   * The VPC connection to use for QuickSight to connect to VPC resources
   */
  readonly quickSightVpcConnection: quicksight.CfnVPCConnection;
  /**
   * The ARN of the folder to place the data sources in
   */
  readonly folderArn: string;
  /**
   * The credentials secret's full ARN needed for the redshift workgroup
   */
  readonly redshiftSecretArn: string;
  /**
   * The RenovoLive database endpoint to use for the SQL Server data source
   */
  readonly sqlServerEndpoint: RLDBEndpoints;
  /**
   * The credentials secret's full ARN needed for the SQL Server data source
   */
  readonly sqlServerSecretArn: RLDBCredSecretArns;
}

export class RenovoLiveDataSources extends Stack {
  /**
   * The Redshift data source for RL
   */
  public readonly redshiftDataSource: quicksight.CfnDataSource;
  /**
   * The SQL Server data source for RL
   */
  public readonly sqlServerDataSource: quicksight.CfnDataSource

  constructor(scope: Construct, id: string, props: RenovoLiveDataSourcesProps) {
    super(scope, id, props);

    this.redshiftDataSource = new quicksight.CfnDataSource(this, 'RedshiftDataSource', {
      awsAccountId: this.account,
      dataSourceId: `${props.renovoLiveEnv}-renovolive-redshift`,
      name: `${props.renovoLiveEnv}-renovolive-redshift`,
      type: 'REDSHIFT',
      dataSourceParameters: {
        redshiftParameters: {
          database: 'renovolive',
          host: props.redshiftWorkgroup.attrWorkgroupEndpointAddress,
          port: 5439,
        },
      },
      credentials: {
        secretArn: props.redshiftSecretArn,
      },
      folderArns: [
        props.folderArn,
      ],
      vpcConnectionProperties: {
        vpcConnectionArn: props.quickSightVpcConnection.attrArn,
      },
    });

    this.sqlServerDataSource = new quicksight.CfnDataSource(this, 'SqlServerDataSource', {
      awsAccountId: this.account,
      dataSourceId: `${props.renovoLiveEnv}-renovolive-sqlserver`,
      name: `${props.renovoLiveEnv}-renovolive-sqlserver`,
      type: 'SQLSERVER',
      dataSourceParameters: {
        sqlServerParameters: {
          database: 'RenovoMaster',
          host: props.sqlServerEndpoint,
          port: 1433,
        },
      },
      credentials: {
        secretArn: props.sqlServerSecretArn,
      },
      folderArns: [
        props.folderArn,
      ],
      vpcConnectionProperties: {
        vpcConnectionArn: props.quickSightVpcConnection.attrArn
      },
    });
  }
}