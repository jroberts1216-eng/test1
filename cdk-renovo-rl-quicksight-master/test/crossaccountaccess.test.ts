import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { RenovoLiveQuickSightCrossAccountAccessStack } from '../src/crossaccountqsaccess';

test('Snapshot', () => {
  const app = new App();

  const renovoLiveQuickSightCrossAccountAccess = new RenovoLiveQuickSightCrossAccountAccessStack(app, 'QuickSightCrossAccountAccessRenovoLive', {
    env: {
      account: '123456789012',
      region: 'us-east-1',
    },
    accountIds: ['123456789013', '123456789014'],
    externalId: 'ab1c723d-45e6-789f-ghi1-4j23k4l56m78',
    roleName: 'renovolive-quicksight-cross-account-role',
  });

  const renovoLiveQuickSightCrossAccountAccessTemplate = Template.fromStack(renovoLiveQuickSightCrossAccountAccess);
  expect(renovoLiveQuickSightCrossAccountAccessTemplate.toJSON()).toMatchSnapshot();
});