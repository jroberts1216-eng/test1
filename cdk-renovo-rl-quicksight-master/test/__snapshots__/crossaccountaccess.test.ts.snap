// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "QuickSightCrossAccountPolicy8D929E90": {
      "Metadata": {
        "cdk_nag": {
          "rules_to_suppress": [
            {
              "id": "AwsSolutions-IAM5",
              "reason": "This policy is intended to be used with cross-account access to QuickSight and the wildcarded permissions are necessary for the use case.         See role policy guidance here: https://aws.amazon.com/blogs/modernizing-with-aws/embedding-amazon-quicksight-analytics-in-net-applications/",
            },
          ],
        },
      },
      "Properties": {
        "Description": "Policy to be used with cross-account access to QuickSight",
        "Path": "/",
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "quicksight:ListUsers",
                "quicksight:DescribeUser",
                "quicksight:RegisterUser",
                "quicksight:ListDashboards",
                "quicksight:DescribeDashboard",
                "quicksight:ListDashboardVersions",
                "quicksight:GetSessionEmbedUrl",
                "quicksight:CreateGroupMembership",
                "quicksight:DeleteGroupMembership",
                "quicksight:DeleteUser",
              ],
              "Effect": "Allow",
              "Resource": "*",
              "Sid": "Default",
            },
            {
              "Action": [
                "quicksight:GenerateEmbedUrlForRegisteredUser",
                "quicksight:GetAuthCode",
                "quicksight:SearchDashboards",
              ],
              "Effect": "Allow",
              "Resource": "arn:aws:quicksight:us-east-1:123456789012:user/*",
              "Sid": "AllowEmbedForRegisteredUser",
            },
            {
              "Action": "quicksight:SearchDashboards",
              "Effect": "Allow",
              "Resource": "arn:aws:quicksight:us-east-1:123456789012:dashboard/*",
              "Sid": "AllowSearchDashboards",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::ManagedPolicy",
    },
    "QuickSightCrossAccountRoleB6A3E0A2": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Condition": {
                "StringEquals": {
                  "sts:ExternalId": "ab1c723d-45e6-789f-ghi1-4j23k4l56m78",
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::123456789013:root",
                    ],
                  ],
                },
              },
            },
            {
              "Action": "sts:AssumeRole",
              "Condition": {
                "StringEquals": {
                  "sts:ExternalId": "ab1c723d-45e6-789f-ghi1-4j23k4l56m78",
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::123456789014:root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Ref": "QuickSightCrossAccountPolicy8D929E90",
          },
        ],
        "RoleName": "renovolive-quicksight-cross-account-role",
      },
      "Type": "AWS::IAM::Role",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
